client/configuration_set.go

```diff
@@ -0,0 +1,101 @@
+package client
+
+type CreateConfigurationSetPayload struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	// if Scope is "organization", scopeId will be calculated in the functions.
+	Scope                   string                  `json:"scope"`   // "project" or "organization".
+	ScopeId                 string                  `json:"scopeId"` // project id or organization id.
+	ConfigurationProperties []ConfigurationVariable `json:"configurationProperties"`
+}
+
+type UpdateConfigurationSetPayload struct {
+	Name                           string                  `json:"name"`
+	Description                    string                  `json:"description"`
+	ConfigurationPropertiesChanges []ConfigurationVariable `json:"configurationPropertiesChanges"` // delta changes.
+}
+
+type ConfigurationSet struct {
+	Id string `json:"id"`
+}
+
+func (client *ApiClient) ConfigurationSetCreate(payload *CreateConfigurationSetPayload) (*ConfigurationSet, error) {
+	var result ConfigurationSet
+	var err error
+
+	if payload.Scope == "organization" {
+		payload.ScopeId, err = client.OrganizationId()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if err := client.http.Post("/configuration-sets", payload, &result); err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+func (client *ApiClient) ConfigurationSetUpdate(id string, payload *UpdateConfigurationSetPayload) (*ConfigurationSet, error) {
+	var result ConfigurationSet
+
+	if err := client.http.Put("/configuration-sets/"+id, payload, &result); err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+func (client *ApiClient) ConfigurationSets(scope string, scopeId string) ([]ConfigurationSet, error) {
+	var result []ConfigurationSet
+	var err error
+
+	if scope == "organization" {
+		scopeId, err = client.OrganizationId()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if err := client.http.Get("/configuration-sets", map[string]string{
+		"scope":               scope,
+		"scopeId":             scopeId,
+		"includeHigherScopes": "false",
+	}, &result); err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (client *ApiClient) ConfigurationSet(id string) (*ConfigurationSet, error) {
+	var result ConfigurationSet
+
+	if err := client.http.Get("/configuration-sets/"+id, nil, &result); err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+func (client *ApiClient) ConfigurationSetDelete(id string) error {
+	return client.http.Delete("/configuration-sets/"+id, nil)
+}
+
+func (client *ApiClient) ConfigurationVariablesBySetId(setId string) ([]ConfigurationVariable, error) {
+	var result []ConfigurationVariable
+
+	organizationId, err := client.OrganizationId()
+	if err != nil {
+		return nil, err
+	}
+
+	if err := client.http.Get("/configuration", map[string]string{
+		"setId":          setId,
+		"organizationId": organizationId,
```

Comment:
don't send the `organizationId` it will return both the variables of the set and the organization...
You should send only the `setId`

Reply:
got it! thanks!